### PR TITLE
Aggregate results

### PR DESCRIPTION
**Pull request by jacobvjk**
closes ADO 4309

**This PR**
- aggregates credit risk results to the portfolio/sector level by using the exposure for weighting instead of sector level with wrong equity weighting
  - this means we now have `<asset_type>_portfolio_pd_changes_annual_<xyz>` and `<asset_type>_portfolio_pd_changes_overall_<xyz>` instead of `<asset_type>_sector_pd_changes_annual_<xyz>` and `<asset_type>_sector_pd_changes_overall_<xyz>`.
  - The content of these files changes, as the weighting logic changes and the overall structure of the files is more in line with how aggregations for the market risk calculations are presented
- moves all aggregation from the most granular level to the required levels for the output files to the function aggregate_results()

**Expectation**
- This should change the output for portfolio level credit risk results

### CODE REVIEW

**Comment by jacobvjk** [comment]
having fixed the missing exposure issue in #357 
we now get the same results for all files except the pd aggregations that were expected to change from sector level to portfolio level

**Comment by MirijaHa** [comment]
Will review during the day. 

1. On first look I noticed that vignettes where not changed to account for the new results. (When) are you planning to do that?
2. Do yo understand the changes in enough detail @jakubcervenka  to assess if any changes affect the visualisations @MonikaFu is doing? In that case some requirements for the visualisations will have to be adjusted and we will have to provide new toy data.

File: R/aggregate_results.R [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+        "investor_name", "portfolio_name", "company_name",
+        "scenario_geography", "ald_sector"
+      )
+    )
+
+  return(data)
+}
+
+aggregate_pd_change_to_sector_level <- function(data,
+                                                horizon = c("annual", "overall")) {
```
Comment (MirijaHa) [inline R/aggregate_results.R:216]: maybe add a check that horizon is of length 1, this design pattern could be misunderstoof for a default.

File: R/aggregate_results.R [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+#' @param results_list A list of results.
+#' @param sensitivity_analysis_vars  String vector holding names of iteration
+#'   arguments.
+#'
+#' @return A list of including basic and aggregated results.
+aggregate_results <- function(results_list, sensitivity_analysis_vars) {
+  sensitivity_analysis_vars <- paste0(sensitivity_analysis_vars, "_arg")
+
+  # validate value changes on company technology level -------------------------
+  validate_data_has_expected_cols(
```
[the remark this answers is not in the document]
Comment (jacobvjk) [reply R/aggregate_results.R:15]: done

File: R/aggregate_results.R [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+
+  company_value_changes <- data %>%
+    dplyr::select(-c(.data$plan_carsten, .data$plan_sec_carsten, .data$year))
+
+  # Credit Risk
+  # prepare exposure data for credit risk results
+  exposure_at_default_credit <- results_list$exposure_by_technology_and_company %>%
+    prepare_exposure_for_credit_risk_format()
+
+  # TODO: needs a check which lines have been removed
```
[the remark this answers is not in the document]
Comment (jacobvjk) [reply R/aggregate_results.R:54]: not needed anymore, this is recorded in reading the pacta data and first generating the exposure data

File: R/aggregate_results.R [hunk 4]
```diff
@@ -0,0 +1,240 @@
+#' Aggregate results
+#'
+#' Function aggregates results to the expected levels. List element entry
```
[the remark this answers is not in the document]
Comment (jacobvjk) [reply R/aggregate_results.R:3]: done

File: R/aggregate_results.R [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    prepare_exposure_for_credit_risk_format()
+
+  # TODO: needs a check which lines have been removed
+  company_pd_changes_annual <- results_list$company_pd_changes_annual %>%
+    join_credit_exposure_on_pd_changes(exposure_at_default_credit)
+
+  company_pd_changes_annual <- company_pd_changes_annual %>%
+    aggregate_pd_change_to_sector_level(horizon = "annual")
+
+  # TODO: needs a check which lines have been removed
```
[the remark this answers is not in the document]
Comment (jacobvjk) [reply R/aggregate_results.R:61]: s.a.

File: R/company_asset_value_at_risk.R [hunk 6]
```diff
@@ -9,26 +9,17 @@
 #'   shock scenario at hand
```
[the remark this answers is not in the document]
Comment (jacobvjk) [reply R/company_asset_value_at_risk.R:9]: done

**Comment by jacobvjk** [comment]
> Will review during the day.
> 
> 1. On first look I noticed that vignettes where not changed to account for the new results. (When) are you planning to do that @jacobvjk ?
> 2. Where the expectations met? (expected changes on portfolio level creditrisk, no changes on any other of the 7 result files?)
> 3. Do yo understand the changes in enough detail @jakubcervenka  to assess if any changes affect the visualisations @MonikaFu is doing? If it does some requirements for the visualisations will have to be adjusted and we will have to provide new toy data.

re vignette - now updated
re expectations - yes met, see my comment from yesterday :)

**Review by MirijaHa — APPROVED** [review]
lgtm

**Pull request merged**